posthog/session_recordings/session_recording_helpers.py

```diff
@@ -202,10 +206,35 @@ def new_event(items: list[dict] | None = None) -> Event:
                 event = new_event(other_snapshots)
                 yield event
             else:
-                # If not, send them individually
-                for snapshot_data in other_snapshots:
-                    event = new_event([snapshot_data])
-                    yield event
+                # If not, keep splitting the list until all parts fit within headroom
+                # we want to avoid sending them all individually if we can - there could be tens of thousands
+                # in data from these older clients that batched poorly
+                parts = [other_snapshots]
+                loop_count = 0
+                while parts and loop_count < 10:
```

Comment:
Maybe make this a constant that makes it clear it is an arbitrary number that can be played with